Brush up modbus docs (#3478)
OttoWinter/esphomedocs: components/sensor/modbus_controller.rst

- **name** (**Required**, string): The name of the sensor.
- **register_type** (**Required**): type of the modbus register.

>     - ``coil``: Coils are 1-bit registers (ON/OFF values) that are used to control discrete outputs. Read and Write access. Modbus *Function Code 1 (Read Coil Status)* will be used.

Reviewer: ```rst
Modbus *Function Code 1 (Read Coil Status)* will be used.
```

Its a bit inconsistent between the different pages how the register types are explained...

Author: done

>     - ``coil``: Coils are 1-bit registers (ON/OFF values) that are used to control discrete outputs. Read and Write access. Modbus *Function Code 1 (Read Coil Status)* will be used.

Reviewer 2: What Jesse said. Also:
```suggestion
    - ``coil``: Coils are 1-bit registers (ON/OFF values) that are used to control discrete outputs. They may be read and/or written. Modbus *Function Code 1 (Read Coil Status)* will be used.
```

    - ``discrete_input``: discrete input register (read only coil) are similar to coils but can only be read. Modbus *Function Code 2 (Read Input Status)* will be used.

>     - ``holding``: Holding Registers - Holding registers are the most universal 16-bit register. Read and Write access. Modbus *Function Code 3 (Read Holding Registers)* will be used.

Reviewer 2: ```suggestion
    - ``holding``: Holding Registers - Holding registers are the most universal 16-bit register. They may be read and/or written. Modbus *Function Code 3 (Read Holding Registers)* will be used.
```

    - ``FP32``: 32 bit IEEE 754 floating point from 2 registers
    - ``FP32_R``: 32 bit IEEE 754 floating point - same as FP32 but low word first

- **bitmask** (*Optional*, int): sometimes multiple values are packed in a single register's response. The bitmask can be used to extract a value from the response. See :ref:`bitmasks`.

> - **skip_updates** (*Optional*, int): By default all sensors of a modbus_controller are updated together. For data points that don't change very frequently updates can be skipped. A value of 5 would only update this sensor range in every 5th update cycle. Note: The modbus_controller groups components by address ranges to reduce number of transactions. All components with the starting same address will be updated in one request. ``skip_updates`` applies for *all* components in the same range.

Reviewer 2: ```suggestion
- **skip_updates** (*Optional*, int): By default, all sensors of a modbus_controller are updated together. For data points that don't change very frequently, updates can be skipped. A value of 5 would only update this sensor range in every 5th update cycle. Note: The modbus_controller groups components by address ranges to reduce number of transactions. All components with the same starting address will be updated in one request. ``skip_updates`` applies for *all* components in the same range.
```

> - **register_count** (*Optional*, int): Number of consecutive registers  this data point spans or to skip in a single read command. Default is 1. See :ref:`modbus_register_count` for more details.

Reviewer 2: This reads weirdly. Maybe this?:
```suggestion
- **register_count** (*Optional*, int): The number of consecutive registers this read request should span or skip in a single command. Default is 1. See :ref:`modbus_register_count` for more details.
```

   - ``return <FLOATING_POINT_NUMBER>;`` the new value for the sensor.
   - ``return NAN;`` if the state should be considered invalid to indicate an error (advanced).

- **custom_command** (*Optional*, list of bytes): raw bytes for modbus command. This allows using non-standard commands. If ``custom_command`` is used ``address`` and ``register_type`` can't be used.

>   custom data must contain all required bytes including the modbus device address. The crc is automatically calculated and appended to the command.

Reviewer 2: ```suggestion
  Custom data must contain all required bytes including the modbus device address. The CRC is automatically calculated and appended to the command.
```
